docs/techniques/exposing-polymorphic-types-swagger.md

```diff
@@ -0,0 +1,71 @@
+# Type inheritance support in Swagger API
+
+Thanks to [PR #2063](https://github.com/VirtoCommerce/vc-platform/pull/2063), Virtocommerce platform is able to expose derived types in Swagger API description. 
+
+## Problem
+Consider the following code:
+```csharp
+// Models definition in ...Core assembly
+
+public abstract class BaseObject
+{
+    // ...
+}
+
+public class DerivedObject : BaseObject
+{
+    // ...
+}
+
+public class AnotherDerivedObject : BaseObject
+{
+    // ...
+}
+
+// Controller method that returns these models
+public ActionResult<IList<BaseObject>> GetObjects()
+{
+    var result = new[] 
+    {
+        new DerivedObject(),
+        new AnotherDerivedObject()
+    };
+
+    return Ok(result);
+}
+```
+
+In this example, only the `BaseObject` will be exposed in Swagger. `DerivedObject` and `AnotherDerivedObject` will be missing from JSON with API definition, and the only way to expose them is to add some other API method that will explicitly accept or return any of these derived types.
+
+On the other hand, we don't want to enable polymorphism globally, as this might break existing API (e.g. any action from `vc-module-customer` that works with a `Member` class).
+
+## Solution
+Since the creation of VC platform v3, polymorphism support in Swashbuckle [improved significantly](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/1792). Now we actually can use it in actions, and the resulting document will be suitable for AutoRest (e.g. to generate API clients for storefront). An example mentioned above can be reworked like this to expose derived models:
+
+```csharp
+using Swashbuckle.AspNetCore.Annotations;
+
+[SwaggerSubType(typeof(DerivedObject)]
+[SwaggerSubType(typeof(AnotherDerivedObject)]
```

Comment:
It might also be nice to also mention [type discriminator annotations](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#enrich-polymorphic-base-classes-with-discriminator-metadata). Unfortunately, this seems to be poorly implemented in Swagger client generation tools, but this might change in the future.